openedx/stanford/pavelib/utils/i18n_helpers.py

```diff
@@ -151,32 +290,73 @@ def fix_metadata(pofile):
-    """
-    from openedx.stanford.pavelib.i18n import CONFIG
-    target_filename = "stanford_" + target + '.po'
+def merge_translations():
+    for language in CONFIG.translated_locales:
+        LOG.info("Merging language: %s...", language)
+        if not CONFIG.get_messages_dir(language).exists():
+            # Language not yet available in code, fetch from upstream Transifex
+            LOG.warn("Fetch upstream translations manually for %s", language)
+            break
+        merge_mappings = {
+            'django.po': ['django', 'mako'],
+            'djangojs.po': ['djangojs', 'underscore'],
+        }
+        for existing, targets in merge_mappings.items():
+            for target in targets:
+                _merge_existing_translations(existing, target, language)
+        LOG.info("Merged language: %s.", language)
 
-    # Fetch common messages first
-    msgcomm_template = 'msgcomm {existing_file} {target_source} -o {output}'
-    target_source = CONFIG.source_messages_dir / target_filename
-    common_file = CONFIG.get_messages_dir(lang) / 'common.po'
-    msgcomm_cmd = msgcomm_template.format(
-        existing_file=CONFIG.get_messages_dir(lang) / existing,
+
+def _merge_common_fetch(target_source, target_filename, common_file, language, existing_file):
+    existing_file = CONFIG.get_messages_dir(lang) / existing
+    command = "msgcomm {existing_file} {target_source} -o {output}".format(
+        existing_file=existing_file,
         target_source=target_source,
         output=common_file,
     )
-    execute(msgcomm_cmd)
+    execute(command)
+
 
-    msgmerge_template = 'msgmerge --no-fuzzy-matching {common_file} {target_source} -o {output}'
-    msgmerge_cmd = msgmerge_template.format(
+def _merge_common_merge(target_source, target_filename, common_file, language):
+    output = CONFIG.get_messages_dir(language) / target_filename
+    command = "msgmerge --no-fuzzy-matching {common_file} {target_source} -o {output}".format(
         common_file=common_file,
         target_source=target_source,
-        output=CONFIG.get_messages_dir(lang) / target_filename,
+        output=output,
+    )
+    execute(command)
+
+
+def _merge_common_push(target, language):
+    command = "tx push -t -l {language} -r stanford-openedx.{target}".format(
+        language=language,
+        target=target,
+    )
+    execute(command)
+
+
+def _merge_existing_translations(existing, target, language):
+    """
+    Merge translations from existing file to target file for locale
+    language and push up to Transifex
+    """
+    target_filename = "stanford_" + target + '.po'
+    target_source = CONFIG.source_messages_dir / target_filename
+    common_file = CONFIG.get_messages_dir(language) / 'common.po'
+    _merge_common_fetch(target_source, target_filename, common_file, language, existing_file)
+    _merge_common_merge(target_source, target_filename, common_file, language)
+    _merge_common_push(target, language)
```

Comment:
I personally don't find much value in the further factoring of these 3 helpers compared with the previous `merge_existing_translations`; it's not a repeated call used by anything else, and the function names are kind of obtuse, and lots of argument passing that is error prone. See comments for fixes to make it work but I feel like the original was easier to parse, just my opinion

Reply:
Actually just changing `_merge_common_push` to `_merge_push` would be clearer, I'm okay with the others, not a huge deal

Reply:
Yeah, the names weren't good and the arguments cargo-culted.

I went ahead and renamed to `_merge_push`, `_merge_pull`, and `_merge_combine`.

I removed/renamed arguments, as requested.

The desire to split those out in refactoring wasn't for reuse, but readability.

Now, pulling up `_merge_existing_translations` reads like:
- initialize some variables
- pull
- combine
- push

which is easier coming into than trying to parse the differences between `msgcomm`, `msgmerge`, `tx push`, etc.

While I'm not planning on doing anything with it now, we do have two different calls to `tx push`, so in theory, those could be combined in the future.